### PR TITLE
feat(dbx-edge-tpa): third party apps sync included

### PR DESCRIPTION
**Pull request by athielba**
## Description

1. Sync third party apps
2. Refresh app
3. Delete App

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/dropbox-edge/src/connectors/dropbox/apps.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  if (!response.ok) {
+    throw await DropboxError.fromResponse('Could not retrieve linked apps', { response });
+  }
+
+  const data: unknown = await response.json();
+
+  const result = linkedAppsSchema.omit({ team_member_id: true }).safeParse(data);
+
+  if (!result.success) {
+    throw await DropboxError.fromResponse('Could not parse linked apps response', { response });
```
Comment (elbalexandre) [inline apps/dropbox-edge/src/connectors/dropbox/apps.ts:93]: Not a fan of this, IMO the `result.error` is crucial to debug and understand which field is wrong / missing.
Happened to me while debugging confluence, having the zod errors logged is really convenient.

File: apps/dropbox-edge/src/connectors/dropbox/users.ts [hunk 2]
```diff
@@ -105,7 +105,7 @@ const teamMemberSchema = z.object({
       display_name: z.string().min(1),
     }),
     status: z.object({
-      '.tag': z.union([z.literal('active'), z.literal('suspended')]),
+      '.tag': z.string(), // z.union([z.literal('active'), z.literal('suspended')]),
```
Comment (elbalexandre) [inline apps/dropbox-edge/src/connectors/dropbox/users.ts:108]: Are sure about this ? Perhaps we can remove the comment if so.

File: apps/dropbox-edge/src/connectors/dropbox/users.ts [hunk 3]
```diff
@@ -172,6 +172,11 @@ export const getUsers = async ({
     const userResult = teamMemberSchema.safeParse(member);
 
     if (userResult.success) {
+      // Only active users are valid
+      if (userResult.data.profile.status['.tag'] !== 'active') {
```
Comment (elbalexandre) [inline apps/dropbox-edge/src/connectors/dropbox/users.ts:176]: [`zFilteredArray`](https://github.com/elba-security/elba-security/pull/336) could have been nice to use here :(

File: apps/dropbox-edge/src/connectors/dropbox/apps.ts (hunk 1 above)
Comment (athielba) [reply apps/dropbox-edge/src/connectors/dropbox/apps.ts:93]: Dropbox return common error in the body, unless we log it properly, it is hard to debug, even I dont want to call another method to log the error but I don't find a better way do it

File: apps/dropbox-edge/src/connectors/dropbox/users.ts (hunk 2 above)
Comment (athielba) [reply apps/dropbox-edge/src/connectors/dropbox/users.ts:108]: Point, I added this while I was testing forgot to remove 🙈

File: apps/dropbox-edge/src/connectors/dropbox/users.ts (hunk 3 above)
Comment (elbalexandre) [reply apps/dropbox-edge/src/connectors/dropbox/users.ts:176]: Yes, it's just a remark

**Pull request merged**